devnet/substrate_bob.Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM parity/polkadot:v0.9.10 AS polkadot
+FROM golang:1.17
+
+ARG key
+ARG CHAIN=cross-client
+ARG DD_API_KEY=somekey
+ARG METRICS_NAMESPACE=substrate.local.devnet
+
+ENV key=${key}
+ENV CHAIN=${CHAIN}
+ENV DD_API_KEY=${DD_API_KEY}
+
+RUN test -n "${key}"
+RUN DD_AGENT_MAJOR_VERSION=7 DD_INSTALL_ONLY=true DD_SITE="datadoghq.com" bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
+
+COPY --from=polkadot /usr/bin/polkadot /usr/bin/polkadot
+
+WORKDIR /gossamer
+
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . . 
+
+WORKDIR /gossamer/devnet
```

Comment:
Careful `/gossamer/devnet` uses a different `go.mod` than the root `go.mod`.

Reply:
When I do `COPY . .` after `RUN go mod download` I'm copying the `devnet` folder as well right? I'm not sure why do I need to copy `go.mod` and `go.sum` before the code?

Reply:
Oh I'm an idiot, this is a Dockerfile in the devnet directory, my bad!



> I'm not sure why do I need to copy `go.mod` and `go.sum` before the code?

For Docker layer caching to have faster rebuilds, since go.mod doesn't change much and go mod download takes some decent amount of time 😉